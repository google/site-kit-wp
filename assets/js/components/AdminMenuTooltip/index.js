export * from './useShowTooltip';
export * from './AdminMenuTooltip';
