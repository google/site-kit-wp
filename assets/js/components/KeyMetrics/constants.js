/**
 * Key Metrics components - constants.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const KEY_METRICS_SETUP_CTA_WIDGET_SLUG = 'key-metrics-setup-cta-widget';

export const KEY_METRICS_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-key-metrics-selection-panel-opened';
export const KEY_METRICS_SELECTION_FORM = 'key-metrics-selection-form';
export const KEY_METRICS_SELECTED = 'key-metrics-selected';
export const EFFECTIVE_SELECTION = 'key-metrics-effective-selection';
export const UNSTAGED_SELECTION = 'key-metrics-unstaged-selection';
export const MIN_SELECTED_METRICS_COUNT = 2;
export const MAX_SELECTED_METRICS_COUNT = 4;
export const MAX_SELECTED_METRICS_COUNT_WITH_CONVERSION_EVENTS = 8;
export const KEY_METRICS_GROUP_CURRENT = {
	SLUG: 'current-selection',
	LABEL: __( 'Current selection', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_SUGGESTED = {
	SLUG: 'suggested',
	LABEL: __( 'Suggested', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_VISITORS = {
	SLUG: 'visitors',
	LABEL: __( 'Visitors', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_DRIVING_TRAFFIC = {
	SLUG: 'driving-traffic',
	LABEL: __( 'Driving traffic', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_GENERATING_LEADS = {
	SLUG: 'generating-leads',
	LABEL: __( 'Generating leads', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_SELLING_PRODUCTS = {
	SLUG: 'selling-products',
	LABEL: __( 'Selling products', 'google-site-kit' ),
};
export const KEY_METRICS_GROUP_CONTENT_PERFORMANCE = {
	SLUG: 'content-performance',
	LABEL: __( 'Content performance', 'google-site-kit' ),
};
