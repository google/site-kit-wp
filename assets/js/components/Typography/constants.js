/**
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const TYPE_DISPLAY = 'display';
export const TYPE_HEADLINE = 'headline';
export const TYPE_TITLE = 'title';
export const TYPE_BODY = 'body';
export const TYPE_LABEL = 'label';

export const VALID_TYPES = [
	TYPE_BODY,
	TYPE_DISPLAY,
	TYPE_HEADLINE,
	TYPE_LABEL,
	TYPE_TITLE,
];

export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export const VALID_SIZES = [ SIZE_SMALL, SIZE_MEDIUM, SIZE_LARGE ];
