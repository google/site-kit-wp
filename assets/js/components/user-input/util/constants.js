/**
 * User Input Answers.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const USER_INPUT_QUESTIONS_PURPOSE = 'purpose';
export const USER_INPUT_QUESTION_POST_FREQUENCY = 'postFrequency';
export const USER_INPUT_QUESTIONS_GOALS = 'goals';

export const USER_INPUT_QUESTIONS_LIST = [
	USER_INPUT_QUESTIONS_PURPOSE,
	USER_INPUT_QUESTION_POST_FREQUENCY,
	USER_INPUT_QUESTIONS_GOALS,
];

export const USER_INPUT_MAX_ANSWERS = {
	[ USER_INPUT_QUESTIONS_PURPOSE ]: 1,
	[ USER_INPUT_QUESTION_POST_FREQUENCY ]: 1,
	[ USER_INPUT_QUESTIONS_GOALS ]: 3,
};

export const USER_INPUT_CURRENTLY_EDITING_KEY =
	'googlesitekit-user-input-currently-editing';

export const FORM_USER_INPUT_QUESTION_NUMBER = 'user_input_question_number';
export const FORM_USER_INPUT_QUESTION_SNAPSHOT = 'user_input_question_snapshot';
export const USER_INPUT_LEGACY_SITE_PURPOSE_DISMISSED_ITEM_KEY =
	'user-input-legacy-site-purpose-dismissed-item';

/**
 * Gets available questions for user input settings.
 *
 * @since 1.139.0
 * @private
 *
 * @return {Object} Questions object.
 */
export function getUserInputQuestions() {
	return [
		{
			title: __(
				'What is the main purpose of this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answer, Site Kit will tailor the metrics you see on your dashboard to help you track progress towards your specific goals',
				'google-site-kit'
			),
		},
		{
			title: __(
				'How often do you create new content for this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answer, Site Kit will suggest new features for your dashboard related to content creation',
				'google-site-kit'
			),
		},
		{
			title: __(
				'What are your top goals for this site?',
				'google-site-kit'
			),
			description: __(
				'Based on your answers, Site Kit will tailor the metrics and recommendations you see on your dashboard to help you make progress in these areas',
				'google-site-kit'
			),
		},
	];
}

/**
 * Gets available answers for user input settings.
 *
 * @since 1.20.0
 * @private
 *
 * @return {Object} Answers object.
 */
export function getUserInputAnswers() {
	return {
		USER_INPUT_ANSWERS_PURPOSE: {
			sell_products_or_service: __(
				'Sell products or services',
				'google-site-kit'
			),
			sell_products: __( 'Sell products', 'google-site-kit' ),
			provide_services: __( 'Provide services', 'google-site-kit' ),
			monetize_content: __(
				'Monetize content (with ads or affiliate links)',
				'google-site-kit'
			),
			publish_blog: __( 'Publish a blog', 'google-site-kit' ),
			publish_news: __( 'Publish news content', 'google-site-kit' ),
			share_portfolio: __(
				'Share a business card or portfolio to represent me or my company online',
				'google-site-kit'
			),
			other: __( 'Other', 'google-site-kit' ),
		},
		USER_INPUT_ANSWERS_POST_FREQUENCY: {
			never: __( 'Never', 'google-site-kit' ),
			daily: __( 'Daily', 'google-site-kit' ),
			weekly: __( 'Weekly', 'google-site-kit' ),
			monthly: __( 'Monthly', 'google-site-kit' ),
			other: __( 'Other', 'google-site-kit' ),
		},
		USER_INPUT_ANSWERS_GOALS: {
			retaining_visitors: __(
				'Retain visitors, turn them into loyal readers or customers',
				'google-site-kit'
			),
			improving_performance: __(
				'Improve speed and performance',
				'google-site-kit'
			),
			finding_new_topics: __(
				'Find new topics to write about that connect with my audience',
				'google-site-kit'
			),
			growing_audience: __( 'Grow my audience', 'google-site-kit' ),
			expanding_business: __(
				'Expand my business into new cities, states or markets',
				'google-site-kit'
			),
			generating_revenue: __(
				'Generate more revenue',
				'google-site-kit'
			),
			help_better_rank: __(
				'Help my content rank in a better position in Google search results',
				'google-site-kit'
			),
			understanding_content_performance: __(
				'Understand which content is performing best',
				'google-site-kit'
			),
			encourage_to_post: __(
				'Tips for generating and posting engaging content updates',
				'google-site-kit'
			),
			other: __( 'Other', 'google-site-kit' ),
		},
	};
}

/**
 * Gets available answer descriptions for user input settings.
 *
 * @since 1.139.0
 * @private
 *
 * @return {Object} Answer descriptions object.
 */
export function getUserInputAnswersDescription() {
	return {
		USER_INPUT_ANSWERS_PURPOSE: {
			sell_products_or_service: __(
				'E.g. selling products like devices, apparel, equipment, etc. or offering services like courses, consulting, tutoring, etc.',
				'google-site-kit'
			),
			sell_products: __(
				'E.g. selling devices, apparel, equipment, etc.',
				'google-site-kit'
			),
			provide_services: __(
				'E.g. offering courses, consulting, tutoring, etc.',
				'google-site-kit'
			),
			monetize_content: __(
				'Using display ads, affiliate links, sponsored content, etc.',
				'google-site-kit'
			),
			publish_blog: __(
				'Writing on a topic youâ€™re passionate about, no focus on monetizing content',
				'google-site-kit'
			),
			publish_news: __(
				'E.g. local news, investigative pieces, interviews, etc.',
				'google-site-kit'
			),
			share_portfolio: __(
				'My website represents me or my company',
				'google-site-kit'
			),
			other: undefined,
		},
	};
}
