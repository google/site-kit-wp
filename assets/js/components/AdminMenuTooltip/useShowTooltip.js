import { useCallback } from '@wordpress/element';

import { useSelect, useDispatch } from 'googlesitekit-data';
import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';

export function useShowTooltip( tooltipSettings ) {
	const { setValue } = useDispatch( CORE_UI );

	const hasMinimumWordPress62 = useSelect( ( select ) =>
		select( CORE_SITE ).hasMinimumWordPressVersion( '6.2' )
	);

	const hasMinimumWordPress64 = useSelect( ( select ) =>
		select( CORE_SITE ).hasMinimumWordPressVersion( '6.4' )
	);

	return useCallback( async () => {
		// Check if the WordPress admin menu is open, and if not, open it.
		// The admin menu is hidden via responsive CSS. This is a simple and effective way to check if it's visible.
		const isAdminMenuOpen =
			document.querySelector( '#adminmenu' ).offsetHeight > 0;

		if ( ! isAdminMenuOpen ) {
			const adminMenuToggle = document.getElementById(
				'wp-admin-bar-menu-toggle'
			);

			if ( adminMenuToggle ) {
				adminMenuToggle.firstChild.click();

				// On iOS, at least, this is necessary, without it the settings menu item
				// is not scrolled into view when the Tooltip is shown.
				await new Promise( ( resolve ) => {
					setTimeout( resolve, 0 );
				} );
			}
		}

		// Check if the Site Kit admin submenu is hidden, and if so, show it.
		const adminSubMenuSelector =
			"#adminmenu [href*='page=googlesitekit-dashboard']";
		const isAdminSubMenuHidden = !! document.querySelector(
			`${ adminSubMenuSelector }[aria-haspopup=true]`
		);

		if ( isAdminSubMenuHidden ) {
			document.querySelector( adminSubMenuSelector ).click();
		}

		// This is a hack to prevent the WordPress admin menu from auto-closing when the tooltip takes the focus.
		// This is applicable for WordPress versions 6.2 and 6.3.
		// See https://github.com/WordPress/wordpress-develop/commit/a9fc43e.
		// It's no longer needed from 6.4, see https://github.com/WordPress/wordpress-develop/commit/93cc3b17.
		if ( hasMinimumWordPress62 && ! hasMinimumWordPress64 ) {
			const originalHasFocus = document.hasFocus;
			document.hasFocus = () => {
				document.hasFocus = originalHasFocus;
				return false;
			};
		}

		setValue( 'admin-menu-tooltip', {
			isTooltipVisible: true,
			rehideAdminMenu: ! isAdminMenuOpen,
			rehideAdminSubMenu: isAdminSubMenuHidden,
			...tooltipSettings,
		} );
	}, [
		hasMinimumWordPress62,
		hasMinimumWordPress64,
		setValue,
		tooltipSettings,
	] );
}
