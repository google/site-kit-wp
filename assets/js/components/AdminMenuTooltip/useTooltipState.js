import { useSelect } from 'googlesitekit-data';
import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';

export function useTooltipState() {
	return useSelect(
		( select ) =>
			select( CORE_UI ).getValue( 'admin-menu-tooltip' ) || {
				isTooltipVisible: false,
				rehideAdminMenu: false,
				rehideAdminSubMenu: false,
			}
	);
}
