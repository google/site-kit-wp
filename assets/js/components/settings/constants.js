/**
 * Settings constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { MODULE_SLUG_ADS } from '../../modules/ads/datastore/constants';
import { MODULE_SLUG_SIGN_IN_WITH_GOOGLE } from '../../modules/sign-in-with-google/datastore/constants';
import { MODULE_SLUG_READER_REVENUE_MANAGER } from '../../modules/reader-revenue-manager/datastore/constants';

export const NEW_MODULES = [
	MODULE_SLUG_ADS,
	MODULE_SLUG_READER_REVENUE_MANAGER,
	MODULE_SLUG_SIGN_IN_WITH_GOOGLE,
];

export const BETA_MODULES = [ MODULE_SLUG_SIGN_IN_WITH_GOOGLE ];

export const EXPERIMENTAL_MODULES = [];
