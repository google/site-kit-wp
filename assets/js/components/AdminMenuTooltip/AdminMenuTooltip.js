/**
 * AdminMenuTooltip component.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { useCallback } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { useDispatch, useSelect } from 'googlesitekit-data';
import JoyrideTooltip from '@/js/components/JoyrideTooltip';
import { CORE_UI } from '@/js/googlesitekit/datastore/ui/constants';
import { trackEvent } from '@/js/util';
import useViewContext from '@/js/hooks/useViewContext';
import {
	BREAKPOINT_SMALL,
	BREAKPOINT_TABLET,
	useBreakpoint,
} from '@/js/hooks/useBreakpoint';

export function AdminMenuTooltip() {
	const viewContext = useViewContext();
	const { setValue } = useDispatch( CORE_UI );
	const breakpoint = useBreakpoint();

	const {
		isTooltipVisible = false,
		tooltipSlug,
		title,
		content,
		dismissLabel,
	} = useSelect(
		( select ) =>
			select( CORE_UI ).getValue( 'admin-menu-tooltip' ) || {
				isTooltipVisible: false,
			}
	);

	function handleViewTooltip() {
		trackEvent( `${ viewContext }_${ tooltipSlug }`, 'tooltip_view' );
	}

	const handleDismissTooltip = useCallback( () => {
		// Track dismiss event.
		if ( tooltipSlug ) {
			trackEvent(
				`${ viewContext }_${ tooltipSlug }`,
				'tooltip_dismiss'
			);
		}

		setValue( 'admin-menu-tooltip', undefined );
	}, [ setValue, tooltipSlug, viewContext ] );

	if ( ! isTooltipVisible ) {
		return null;
	}

	const defaultTarget = '#adminmenu [href*="page=googlesitekit-settings"]';
	const isMobileTablet =
		breakpoint === BREAKPOINT_SMALL || breakpoint === BREAKPOINT_TABLET;

	return (
		<JoyrideTooltip
			target={ isMobileTablet ? 'body' : defaultTarget }
			placement={ isMobileTablet ? 'center' : 'right' }
			className={
				isMobileTablet
					? 'googlesitekit-tour-tooltip__modal_step'
					: 'googlesitekit-tour-tooltip__fixed-settings-tooltip'
			}
			disableOverlay={ ! isMobileTablet }
			slug="ga4-activation-banner-admin-menu-tooltip"
			title={ title }
			content={ content }
			dismissLabel={ dismissLabel }
			onView={ handleViewTooltip }
			onDismiss={ handleDismissTooltip }
		/>
	);
}
