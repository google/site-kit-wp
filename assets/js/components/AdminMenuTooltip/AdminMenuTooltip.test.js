/**
 * AdminMenuTooltip component tests.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import {
	act,
	createTestRegistry,
	fireEvent,
	render,
	waitFor,
} from '../../../../tests/js/test-utils';
import { CORE_UI } from '@/js/googlesitekit/datastore/ui/constants';
import { AdminMenuTooltip } from './AdminMenuTooltip';
import * as tracking from '@/js/util/tracking';
import useViewContext from '@/js/hooks/useViewContext';

const mockTrackEvent = vi.spyOn( tracking, 'trackEvent' );
mockTrackEvent.mockImplementation( () => Promise.resolve() );

// Mock useViewContext to return a consistent value
vi.mock( '../../hooks/useViewContext', () => vi.fn() );
useViewContext.mockImplementation( () => 'test-context' );

describe( 'AdminMenuTooltip', () => {
	let registry;

	beforeEach( () => {
		mockTrackEvent.mockClear();
		registry = createTestRegistry();
		vi.useFakeTimers();
	} );

	it( 'should not render when isTooltipVisible is false', async () => {
		registry.dispatch( CORE_UI ).setValue( 'admin-menu-tooltip', {
			isTooltipVisible: false,
		} );

		render(
			<div>
				<div id="adminmenu">
					<a href="http://test.test/wp-admin/admin.php?page=googlesitekit-settings">
						Settings
					</a>
				</div>
				<AdminMenuTooltip />
			</div>,
			{ registry }
		);

		await waitFor( () => {
			expect(
				document.querySelector( '.googlesitekit-tour-tooltip' )
			).not.toBeInTheDocument();
		} );
	} );

	it( 'should render when isTooltipVisible is true and track view event', async () => {
		const tooltipSlug = 'test-tooltip-slug';
		await registry.dispatch( CORE_UI ).setValue( 'admin-menu-tooltip', {
			isTooltipVisible: true,
			title: 'Test Title',
			content: 'Test Content',
			dismissLabel: 'Got it',
			tooltipSlug,
		} );

		render(
			<div className="googlesitekit-plugin">
				<div id="adminmenu">
					<a href="http://test.test/wp-admin/admin.php?page=googlesitekit-settings">
						Settings
					</a>
				</div>
				<AdminMenuTooltip />
			</div>,
			{ registry }
		);

		// Wait for Joyride tooltip's useInterval to render.
		act( () => {
			vi.advanceTimersByTime( 1000 );
		} );

		await waitFor( () => {
			const tooltip = document.querySelector(
				'.googlesitekit-tour-tooltip'
			);
			expect( tooltip ).toBeInTheDocument();
		} );

		expect(
			document.querySelector( '.googlesitekit-tour-tooltip' )
		).toMatchSnapshot();

		// The tracking event should fire when the tooltip is viewed.
		expect( mockTrackEvent ).toHaveBeenCalledWith(
			`test-context_${ tooltipSlug }`,
			'tooltip_view'
		);
	} );

	it( 'should close the tooltip on clicking the `X` button and track dismiss event', async () => {
		const tooltipSlug = 'test-tooltip-slug';
		await registry.dispatch( CORE_UI ).setValue( 'admin-menu-tooltip', {
			isTooltipVisible: true,
			title: 'Test Title',
			content: 'Test Content',
			dismissLabel: 'Got it',
			tooltipSlug,
		} );

		render(
			<div className="googlesitekit-plugin">
				<div id="adminmenu">
					<a href="http://test.test/wp-admin/admin.php?page=googlesitekit-settings">
						Settings
					</a>
				</div>
				<AdminMenuTooltip />
			</div>,
			{ registry }
		);

		// Wait for Joyride tooltip's useInterval to render.
		act( () => {
			vi.advanceTimersByTime( 1000 );
		} );

		let closeButton;
		await waitFor( () => {
			const tooltip = document.querySelector(
				'.googlesitekit-tour-tooltip'
			);
			expect( tooltip ).toBeInTheDocument();

			closeButton = tooltip.querySelector(
				'.googlesitekit-tooltip-close'
			);
			expect( closeButton ).toBeInTheDocument();
		} );

		fireEvent.click( closeButton );

		await waitFor( () => {
			expect(
				document.querySelector( '.googlesitekit-tour-tooltip' )
			).not.toBeInTheDocument();
		} );

		expect(
			registry.select( CORE_UI ).getValue( 'admin-menu-tooltip' )
		).toBeUndefined();

		expect( mockTrackEvent ).toHaveBeenCalledWith(
			`test-context_${ tooltipSlug }`,
			'tooltip_dismiss'
		);
	} );

	it( 'should close the tooltip on clicking the `Got it` button and track dismiss event', async () => {
		const tooltipSlug = 'test-tooltip-slug';
		await registry.dispatch( CORE_UI ).setValue( 'admin-menu-tooltip', {
			isTooltipVisible: true,
			title: 'Test Title',
			content: 'Test Content',
			dismissLabel: 'Got it',
			tooltipSlug,
		} );

		render(
			<div className="googlesitekit-plugin">
				<div id="adminmenu">
					<a href="http://test.test/wp-admin/admin.php?page=googlesitekit-settings">
						Settings
					</a>
				</div>
				<AdminMenuTooltip />
			</div>,
			{ registry }
		);

		// Wait for Joyride tooltip's useInterval to render.
		act( () => {
			vi.advanceTimersByTime( 1000 );
		} );

		let gotItButton;
		await waitFor( () => {
			const tooltip = document.querySelector(
				'.googlesitekit-tour-tooltip'
			);
			expect( tooltip ).toBeInTheDocument();

			gotItButton = tooltip.querySelector(
				'.googlesitekit-tooltip-button'
			);
			expect( gotItButton ).toBeInTheDocument();
		} );

		// Click the Got it button
		fireEvent.click( gotItButton );

		await waitFor( () => {
			expect(
				document.querySelector( '.googlesitekit-tour-tooltip' )
			).not.toBeInTheDocument();
		} );

		expect(
			registry.select( CORE_UI ).getValue( 'admin-menu-tooltip' )
		).toBeUndefined();

		expect( mockTrackEvent ).toHaveBeenCalledWith(
			`test-context_${ tooltipSlug }`,
			'tooltip_dismiss'
		);
	} );
} );
