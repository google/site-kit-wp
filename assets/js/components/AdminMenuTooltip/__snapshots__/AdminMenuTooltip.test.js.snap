// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMenuTooltip should render when isTooltipVisible is true and track view event 1`] = `
<div
  aria-modal="true"
  class="googlesitekit-tour-tooltip"
  role="alertdialog"
>
  <div
    class="mdc-card googlesitekit-tooltip-card"
  >
    <div
      class="googlesitekit-tooltip-body"
    >
      <h2
        class="googlesitekit-tooltip-title"
      >
        Test Title
      </h2>
      <div
        class="googlesitekit-tooltip-content"
      >
        Test Content
      </div>
    </div>
    <div
      class="mdc-card__actions googlesitekit-tooltip-actions"
    >
      <ul
        class="googlesitekit-tooltip-indicators"
      />
      <div
        class="googlesitekit-tooltip-buttons"
      >
        <button
          aria-label="Got it"
          class="mdc-button googlesitekit-tooltip-button"
          data-action="primary"
          role="button"
          target="_self"
        >
          <span
            class="mdc-button__label"
          >
            Got it
          </span>
        </button>
      </div>
    </div>
    <button
      aria-label="Close"
      class="mdc-button googlesitekit-tooltip-close"
      target="_self"
    >
      <svg />
    </button>
  </div>
</div>
`;
